docs: Oracle Audit Logs (#32037)
gravitational/teleport: docs/pages/database-access/guides/oracle-self-hosted.mdx

teleport-kube-agent-0   1/1     Running   0          32s
```

> ## Step 5/6. Configure Teleport Oracle Audit Logs Puller (Optional).

Reviewer: ```suggestion
## Step 5/6. (Optional) Configure Teleport to pull audit logs from Oracle Audit Trail
```
"Oracle Audit Logs Puller" isn't a Teleport product, so I've suggested a more descriptive heading

Teleport can pull audit logs from Oracle Audit Trail.
In order to enable this feature, you will need to configure Oracle Audit Trail and create a dedicated Teleport
user that will be used to fetch audit events from Oracle Audit Trail.

> Create a internal Oracle `teleport` user that will be used as db user that will be used to fetch
> audit events from Oracle Audit Trail:

Reviewer: ```suggestion
Create an internal Oracle `teleport` user that will fetch
audit events from Oracle Audit Trail:
```

```sql
CREATE USER teleport IDENTIFIED EXTERNALLY AS 'CN=teleport';
GRANT CREATE SESSION TO teleport;
GRANT SELECT ON dba_audit_trail TO teleport;
GRANT SELECT ON V_$SESSION TO teleport;
```

> Enable table Oracle Audit Trail:

Reviewer: ```suggestion
Enable the table in Oracle Audit Trail:

```

```
ALTER system SET audit_trail=db,extended scope=spfile;
```

> <Admonition type="info" title="audit_trait">
> In order to propagate audit_trail changes, Oracle instance needs to be restarted.
> </Admonition>

Reviewer: ```suggestion
Restart your Oracle instance to propagate audit trail changes.
```

Since this is a part of the flow we're describing, we should put it in the main body text rather than an `Admonition`.

> Enable Audit Oracle activity for the `alice` user:

Reviewer: ```suggestion
Enable Oracle auditing for the `alice` user:

```
Is "Audit Oracle" a proper noun? I've suggested another wording.

```sql
AUDIT ALL BY alice BY ACCESS;

> AUDIT SELECT TABLE, UPDATE TABLE, INSERT TABLE, DELETE TABLE BY alice BY ACCESS;

Reviewer 2: If my understanding is correct, there are a few more statements not covered by `ALL` like `ALTER TABLE`, `GRANT PROCEDURE`?

Maybe leave a link in the next steps to learn about AUDIT?

I also wonder if `AUDIT ALL STATEMENTS` would help in this case.

```

> <Admonition type="info" title="Audit Oracle user activity ">
> Enabling Audit needs to be done for each user Teleport user that will be used to connect to Oracle.
> Additionally you create different audit policy for each user.
> </Admonition>

Reviewer: ```suggestion
You must enable auditing for each Teleport user that will be used to connect to Oracle.
Additionally you can create a different audit policy for each user.
```

> Configure Teleport DB Service to pull audit logs from Oracle Audit Trail:

Reviewer: ```suggestion
Configure the Teleport Database Service to pull audit logs from Oracle Audit Trail:
```

<Tabs>
<TabItem label="Static config">
```yaml
db_service:
  enabled: "yes"
  databases:
  - name: "oracle"
    protocol: "oracle"
    uri: oracle.example.com:2484"
    oracle:
      audit_user: "teleport"
```
</TabItem>
<TabItem label="Dynamic resource">
```yaml
kind: db
version: v3
metadata:
  name: oracle
spec:
  protocol: "oracle"
  uri: "oracle.example.com:2484"
  oracle:
    audit_user: "teleport"
```
</TabItem>
</Tabs>

<Admonition type="warning" title="Audit Trail Cleanup">
Teleport doesn't clean up audit trail events from Oracle Audit Trail.

> Make sure to configure Oracle Audit Trail cleanup policy to avoid running out of disk space.

Reviewer: Are there any quick commands we can show the user for configuring an audit trail cleanup policy?

> <Admonition type="warning" title="Audit Trail Cleanup">
> Teleport doesn't clean up audit trail events from Oracle Audit Trail.
> Make sure to configure Oracle Audit Trail cleanup policy to avoid running out of disk space.
> </Admonition>

Reviewer: ```suggestion
Teleport doesn't clean up audit trail events from Oracle Audit Trail.
Make sure to configure an Oracle Audit Trail cleanup policy to avoid running out of disk space.
```

I don't think this needs to be an Admonition.
